state/application_test.go

```diff
@@ -2518,37 +2514,25 @@ func (s *ApplicationSuite) TestDestroyWithRemovableApplicationOpenedPortRanges(c
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(portRangesUnit1.UniquePortRanges(), gc.HasLen, 2)
 
-	appPortRanges, err = app.OpenedPortRanges()
-	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(appPortRanges.UniquePortRanges(), gc.HasLen, 3)
-
 	portRangesUnit1.Close(allEndpoints, network.MustParsePortRange("3002/tcp"))
 	c.Assert(st.ApplyOperation(portRangesUnit1.Changes()), jc.ErrorIsNil)
 
 	portRangesUnit1, err = unit1.OpenedPortRanges()
 	c.Assert(err, jc.ErrorIsNil)
 	c.Assert(portRangesUnit1.UniquePortRanges(), gc.HasLen, 1)
 
-	appPortRanges, err = app.OpenedPortRanges()
-	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(appPortRanges.UniquePortRanges(), gc.HasLen, 2)
-
 	portRangesUnit1.Open(allEndpoints, network.MustParsePortRange("3003/tcp"))
 	c.Assert(st.ApplyOperation(portRangesUnit1.Changes()), jc.ErrorIsNil)
 
-	appPortRanges, err = app.OpenedPortRanges()
+	portRangesUnit1, err = unit1.OpenedPortRanges()
 	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(appPortRanges.UniquePortRanges(), gc.HasLen, 3)
+	c.Assert(portRangesUnit1.UniquePortRanges(), gc.HasLen, 2)
 
 	err = unit1.EnsureDead()
 	c.Assert(err, jc.ErrorIsNil)
 	err = unit1.Remove(state.NewObjectStore(c, s.State.ModelUUID()))
 	c.Assert(err, jc.ErrorIsNil)
 
-	appPortRanges, err = app.OpenedPortRanges()
-	c.Assert(err, jc.ErrorIsNil)
-	c.Assert(appPortRanges.UniquePortRanges(), gc.HasLen, 2)
-
```

Comment:
Before each of these removed assertions we already make an equivalent assertion on the unit.

With one exception. In this case, write a unit-based assertion instead